src/editors/config/ProxyEditor.tsx

```diff
@@ -18,32 +19,81 @@ type ProxyEditorProps = DataSourcePluginOptionsEditorProps<InfinityOptions>;
+    onOptionsChange({ ...options, jsonData: { ...jsonData, proxy_username } });
+  };
+  const onResetSecret = (key: keyof InfinitySecureOptions) => {
+    onOptionsChange({
+      ...options,
+      secureJsonFields: { ...options.secureJsonFields, [key]: false },
+      secureJsonData: { ...options.secureJsonData, [key]: '' },
+    });
+  };
   return (
-    <>
-      <div className="gf-form">
-        <InlineLabel width={20}>Proxy</InlineLabel>
+    <Stack gap={0.5} direction={'column'}>
+      <Stack gap={0}>
+        <InlineLabel width={20}>Proxy Mode</InlineLabel>
         <RadioButtonGroup<ProxyType>
           value={jsonData?.proxy_type || 'env'}
           options={[
             { value: 'env', label: 'From environment variable / Default' },
-            { value: 'url', label: 'URL' },
             { value: 'none', label: 'None' },
+            { value: 'url', label: 'URL' },
           ]}
           onChange={(e) => onProxyTypeChange(e || 'env')}
         />
-      </div>
+      </Stack>
       {jsonData?.proxy_type === 'url' && (
-        <div className="gf-form">
-          <InlineLabel width={20}>Proxy URL</InlineLabel>
-          <Input value={jsonData?.proxy_url || ''} onChange={(e) => onProxyUrlChange(e.currentTarget.value)} />
-        </div>
+        <>
+          <Stack gap={0}>
+            <InlineLabel width={20} tooltip={URLSelector.tooltip}>
+              {URLSelector.label}
+            </InlineLabel>
+            <Input
+              role="input"
+              width={40}
+              aria-label={URLSelector.ariaLabel}
+              placeholder={URLSelector.placeholder}
+              value={jsonData?.proxy_url || ''}
+              onChange={(e) => onProxyUrlChange(e.currentTarget.value)}
+            />
+          </Stack>
+          <Stack gap={0}>
+            <InlineLabel width={20} tooltip={UserNameSelector.tooltip}>
+              {UserNameSelector.label}
+            </InlineLabel>
+            <Input
+              role="input"
+              width={40}
+              aria-label={UserNameSelector.ariaLabel}
+              placeholder={UserNameSelector.placeholder}
+              value={jsonData?.proxy_username || ''}
+              onChange={(e) => onProxyUserNameChange(e.currentTarget.value)}
+            />
+          </Stack>
+          <Stack gap={0}>
+            <InlineLabel width={20} tooltip={PasswordSelector.tooltip}>
+              {PasswordSelector.label}
+            </InlineLabel>
+            <SecretInput
+              role="input"
+              width={40}
+              aria-label={PasswordSelector.ariaLabel}
+              placeholder={PasswordSelector.placeholder}
+              isConfigured={(secureJsonFields && secureJsonFields.azureBlobAccountKey) as boolean}
+              onChange={onUpdateDatasourceSecureJsonDataOption(props, 'proxyUserPassword')}
+              onReset={() => onResetSecret('proxyUserPassword')}
+            />
```

Comment:
I think there is something wrong here. When I refresh after configuring the passwod, I got nothing back in the input. And I think `isConfigured={(secureJsonFields && secureJsonFields.azureBlobAccountKey) as boolean}` part is the culprit. Especially `azureBlobAccountKey`